doc/source/toolchain.rst

```diff
@@ -0,0 +1,226 @@
+ 2020              C99
+ ?                 C11
+ ?                 C17, C18
+================  ===========================================
+
+C99 has been a standard for 20 years, but experience has shown that
+not all features are supported equally well across all platforms.
+C18 is a bug fix for C11, so C11 may be skipped entirely.
+
+For C++, we currently suggest that anything beyond C++11 is going to be impossible
+for a very long time because of ecosystem support restrictions. See [4]_.
+
+.. note::
+
+    Developer Note: Some C99 features would be useful for scientific programming, in particular better support of
+    IEEE 754 [5]_.  SciPy has a small include file ``scipy/_lib/_c99compat.h`` which provides
+    access to a few functions.  Use in conjunction with ``<numpy/npy_math.h>``.
+
+    =========================================  ========================================================
+     Feature                                   Workaround
+    =========================================  ========================================================
+    ``isnan()``, ``isinf()``, ``isfinite()``   Use ``sc_isnan()``, ``sc_isinf()``, ``sc_isfinite()``
+    ``NAN``                                    Use ``NPY_NAN`` (*almost* equivalent)
+    inline functions                           Make static functions and place in an include .h file
+    mid-block variable declarations            Declare variables at the top of the block
+    =========================================  ========================================================
+
+
+Fortran Compilers
+^^^^^^^^^^^^^^^^^
+
+Generally, any well-maintained compiler is likely suitable and can be used to build SciPy.
+
+========  ==================
+ Tool      Version
+========  ==================
+gfortran    >= 4.8.0
+ifort      A recent version
+flang      A recent version
+========  ==================
+
+
+Cython Compiler
+^^^^^^^^^^^^^^^
+
+SciPy always requires a recent Cython compiler.
+
+========  ============  ===============
+ Tool     Tool Version   SciPy version                                                z
+========  ============  ===============
+Cython      >= 0.29.1    1.12.1
+========  ============  ===============
+
+
+
+Other Libraries
+^^^^^^^^^^^^^^^
+
+========  ===================
+ Tool      Minimum version
+========  ===================
+LAPACK      3.4.1.
+OpenBLAS    A recent version
+========  ===================
+
+
+Testing and Benchmarking
+--------------------------
+
+Testing and benchmarking require recent versions of:
+
+=========================  ========  ====================================
+ Tool                      Version    URL
+=========================  ========  ====================================
+pytest                     Recent     https://docs.pytest.org/en/latest/
+asv (airspeed velocity)    Recent     https://asv.readthedocs.io/
+mpmath                     Recent     http://mpmath.org
```

Comment:
maybe separate this out in an "optional dependencies" table? and add `scikit-umfpack`?

Reply:
scikit-sparse is an optional dependency of `linprog`(#10026).
(BTW are scikit-sparse and scikit-umfpack not built-in only because SuiteSparse is not BSD?)

Reply:
Indeed, license issue - we cannot include GPL licensed (or similar) code or depend on it directly.

Reply:
Done.